library/hpe_icsp_os_deployment.py

```diff
@@ -203,6 +191,7 @@ def main():
             username=dict(required=True, type='str'),
             password=dict(required=True, type='str', no_log=True),
             server_id=dict(required=True, type='str'),
```

Comment:
I thought about changing this to False, but we need the ip or the serial. If I change the IP to be required=True it would break backward compatibility, then I just check the ip first and if not then use serial number. what do you think?